applications/GeoMechanicsApplication/custom_conditions/U_Pw_face_load_interface_condition.cpp

```diff
@@ -190,26 +190,24 @@ void UPwFaceLoadInterfaceCondition<3,4>::CheckJointWidth(double& rJointWidth, bo
                                                             const double& MinimumJointWidth, const Element::GeometryType& Geom)
 {
     //Quadrilateral_interface_3d_4
-    array_1d<double, 3> pmid0;
-    array_1d<double, 3> pmid1;
-    array_1d<double,3> P2 = Geom.GetPoint( 2 );
-    noalias(pmid0) = 0.5 * (Geom.GetPoint( 0 ) + Geom.GetPoint( 3 ));
-    noalias(pmid1) = 0.5 * (Geom.GetPoint( 1 ) + P2);
+    const auto& P2 = Geom.GetPoint( 2 );
+    const auto& pmid0 = 0.5 * (Geom.GetPoint( 0 ) + Geom.GetPoint( 3 ));
+    const auto& pmid1 = 0.5 * (Geom.GetPoint( 1 ) + P2);
     
     //Unitary vector in local x direction
-    array_1d<double,3> Vx;
-    noalias(Vx) = pmid1 - pmid0;
-    double inv_norm_x = 1.0/norm_2(Vx);
+    const auto& tmpVx = pmid1 - pmid0;
+    array_1d<double,3> Vx{tmpVx[0], tmpVx[1], tmpVx[2]};
```

Comment:
```suggestion
    const array_1d<double,3> Vx = pmid1 - pmid0;
```
Wouldn't it be better to keep it like this? I don't see the point on having the temporary.

Reply:
What about something like this:
`auto Vx = static_cast<array_1d<double,3>>(pmid1 - pmid0);`
to avoid the slicing warnings.

Reply:
Not 100% sure, but I've the feeling that you get the warning because of what is discussed above. I hope that following @huhaas suggestion of removing the `&` would remove the slicing warning too.

Reply:
I have modified this after an internal discussion with @huhaas